atlasdb-api/src/main/java/com/palantir/atlasdb/transaction/api/Transaction.java

```diff
@@ -41,25 +41,95 @@
  */
 public interface Transaction {
 
+    /**
+     * Returns a mapping of rows to {@link RowResult}s within {@code tableRef} for the specified {@code rows}, loading
+     * columns according to the provided {@link ColumnSelection}.
+     *
+     * The returned {@link SortedMap} is sorted on the byte order of row keys; the ordering of the input parameter
+     * {@code rows} is irrelevant. Access to the returned {@link SortedMap} is permitted both with the original byte
+     * array, as well as with byte arrays that are equal in terms of {@link java.util.Arrays#equals(byte[], byte[])}.
+     *
+     * If there are rows with no cells matching the provided {@link ColumnSelection}, they will not be present in the
```

Comment:
It's worth a test for this, I think. Specifically, what do I get back if I e.g. delete a cell in a txn and then read the row?